components/automate-ui/src/app/pages/api-token/details/api-token-details.component.ts

```diff
@@ -98,4 +143,24 @@ export class ApiTokenDetailsComponent implements OnInit, OnDestroy {
   public get nameCtrl(): FormControl {
     return <FormControl>this.updateForm.controls.name;
   }
+
+  // updates whether the project was checked or unchecked
+  onProjectChecked(project: ProjectChecked): void {
+    this.projects[project.id].checked = project.checked;
+    const projectsSelected = Object.values(this.projects).filter(p => p.checked);
+
+    // since the app-projects-dropdown is not a true form input (select)
+    // we have to manage the form reactions
+    this.updateForm.controls.projects.setValue(projectsSelected.map(p => p.id));
+    this.updateForm.controls.projects.markAsDirty();
+  }
+
+  noProjectsUpdated(): boolean {
+    return xor(this.token.projects,
+      Object.values(this.projects).filter(p => p.checked).map(p => p.id)).length === 0;
+  }
+
+  dropdownDisabled(): boolean {
+    return Object.values(this.projects).length === 0 || this.saveInProgress;
```

Comment:
💭 I've commented on this before. How would `this.projects` ever satisfy `Object.values(this.projects).length === 0` unless it was empty, i.e. `{}`? Unless I'm mistaken, lodash has `isEmpty`...
```suggestion
    return isEmpty(this.projects) || this.saveInProgress;
```

Reply:
A team could come in with no projects, in which case it would be empty. And, with bug fix above, you can also now edit the project list to be empty, too.

Reply:
I see. thanks!